src/UIHandler.cpp

```diff
@@ -0,0 +1,345 @@
+#include "UIHandler.h"
+
+UIWindow::UIWindow( int minSizeX, int minSizeY, Location location,
+                    bool drawBorder ) : m_panel( new UIPaddingPanel( drawBorder ) )
+{
+    m_minSize.x = minSizeX;
+    m_minSize.y = minSizeY;
+
+    m_thisLocation = location;
+
+    UpdateWindow();
+}
+
+int UIWindow::UpdateWindow()
+{
+    // TODO: Clear prev contents? Is that our job?
+
+    m_thisSize = m_panel->RequestedSize( Sizes::Prefered );
+
+    point minSize = m_panel->RequestedSize( Sizes::Minimum );
+
+    if( m_minSize.x > m_thisSize.x ) {
+        m_thisSize.x = m_minSize.x;
+    }
+    if( m_minSize.y > m_thisSize.y ) {
+        m_thisSize.y = m_minSize.y;
+    }
+
+    if( m_thisSize.x > TERMX ) {
+        m_thisSize.x = TERMX;
+    }
+    if( m_thisSize.y > TERMY ) {
+        m_thisSize.y = TERMY;
+    }
+
+    if( m_minSize.x > TERMX ) {
+        DebugLog( D_ERROR, DC_ALL ) << "Not enought screen space...";
```

Comment:
If you add debug messages, make them informative: what does "not enough screen space" mean? At least add the values involved in that check.